[#66708] Update filepicker to use Octicons

shiroginne opened the pull request:
# Ticket
[66708](https://community.openproject.org/projects/document-workflows-stream/work_packages/66708)

# What are you trying to accomplish?
Update FilePicker to use Octicons. 

## Screenshots

<img width="666" height="638" alt="Screenshot 2025-09-09 at 12 28 24" src="https://github.com/user-attachments/assets/bbe0555b-2292-456d-bb2e-9938b254d9e0" />
<img width="682" height="639" alt="Screenshot 2025-09-09 at 12 28 31" src="https://github.com/user-attachments/assets/fc1ada4c-9176-41ad-a5af-c7ccfa66e550" />
<img width="962" height="119" alt="Screenshot 2025-09-10 at 10 25 44" src="https://github.com/user-attachments/assets/bdb15a91-cfe1-44bf-9151-5022d100ea9c" />



# Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation in Lookbook (patterns, previews, etc)
- [x] Tested major browsers (Chrome, Firefox, Edge, ...)

shiroginne committed:
[#66708] Change old spot-icons to octicons

- Change links to external instead of android
- Style update
- Disable eslint for label html element

NobodysNightmare commented on frontend/src/app/shared/components/storages/file-link-list-item/file-link-list-item.html:
```diff
@@ -25,9 +25,20 @@
+          <svg *ngSwitchCase="'file'"
+               file-icon
+               size="small"
+               class="op-files-tab--icon op-files-tab--icon_{{fileLinkIcon.clazz}}"></svg>
```
There is a high chance that I am misreading the angular code here, but isn't this case ("file") exactly the same as the default case?

Couldn't we condense the switch-case to "file-media" and default?

Kharonus replied:
I extended the lookup, with default cases.

NobodysNightmare commented on frontend/src/app/shared/components/storages/file-link-list-item/file-link-list-item.html:
```diff
@@ -52,14 +63,20 @@
-          <span class="spot-icon spot-icon_{{action.iconClass}}"></span>
+          <ng-container [ngSwitch]="action.iconClass">
+            <svg *ngSwitchCase="'remove-link'" x-icon size="small" class="octicon"></svg>
+            <span *ngSwitchDefault class="spot-icon spot-icon_{{action.iconClass}}"></span>
```
:yellow_circle: I am surprised to see that we have spot icons left in some cases here. For which actions do we still need spot icons?

Kharonus replied:
`spot-icon` contains a lot of styling, especially styling relative to containers and stuff. transporting this to other classes is more complicated then usefull. :D

NobodysNightmare commented on frontend/src/app/shared/components/storages/storage-file-list-item/storage-file-list-item.html:
```diff
@@ -7,11 +7,16 @@
     data-test-selector="op-files-picker-modal--list-item"
     (click)="enterDirectoryOnLabel($event)"
+    (keydown.enter)="enterDirectoryOnLabel($event)"
+    (keydown.space)="enterDirectoryOnLabel($event)"
```
:red_circle: Are these two lines intentional changes?

I am asking because the PR is about changing icons and this looks like a behavioral change for which I can't find a separate commit introducing it.

shiroginne replied:
Sorry, missed this comment: ES lint was complaining about missing accessibility here, so I had to add it. (code had changed thou, but just for the history)

NobodysNightmare reviewed (commented):
(This is a preliminary review, because I am sinking down into a meeting right now; but that's as far as I came already)

Kharonus commented on frontend/src/app/shared/components/storages/file-link-list-item/file-link-list-item.html:
```diff
@@ -47,19 +49,27 @@
       <ng-container *ngFor="let action of floatingActions">
         <a *ngIf="!!action.href"
-           class="spot-link"
+           class="spot-link octicon-link"
```
🔴 To not leave those classes lying around - and confuse future generations - we should remove the css class `spot-link` here and apply any kind of needed styling to new class you introduced.

Kharonus replied:
I probably need to roll back my own comment here. There is sooooo much styling involved in the `spot` classes, rebuilding that just to get rid of the term in html seems not appropriate. 

Hence, I follow Yauheni's approach of amending the spot styles.

Kharonus commented on frontend/src/app/shared/components/storages/icons.mapping.ts:
```diff
@@ -29,84 +29,84 @@
-  'application/vnd.oasis.opendocument.presentation': { icon: 'file-presentation', clazz: 'presentation' },
-  'application/vnd.oasis.opendocument.presentation-template': { icon: 'file-presentation', clazz: 'presentation' },
-  'application/vnd.apple.keynote': { icon: 'file-presentation', clazz: 'presentation' },
-  'application/vnd.ms-powerpoint': { icon: 'file-presentation', clazz: 'presentation' },
```
🔴 Please do not use `file` for everything. we have several more precise icons in octicons, e.g.:

- op-file-doc
- op-file-sheet
- op-file-csv

and more. Check the [lookbook](https://qa.openproject-edge.com/lookbook/pages/styles/icons) for them ;)

Kharonus commented on frontend/src/app/shared/components/storages/openproject-storages.module.ts:
```diff
@@ -64,12 +64,39 @@ import {
+  LinkExternalIconComponent,
+  UnlinkIconComponent
+} from '@openproject/octicons-angular';
+
```
🔴 Do not import the icons in the storage module, but import them in the `IconModule`. That one is imported already here.

Kharonus commented on frontend/src/app/shared/components/storages/storage-information/storage-information.html:
```diff
@@ -1,4 +1,6 @@
-<span class="spot-icon spot-icon_{{content.iconClass}} info-icon-box"></span>
+<svg *ngIf="content.iconClass === 'warning'" alert-fill-icon size="small" class="info-icon-box"></svg>
+<svg *ngIf="content.iconClass === 'info'" info-icon size="small" class="info-icon-box"></svg>
+<svg *ngIf="content.iconClass !== 'warning' && content.iconClass !== 'info'" question-icon size="small" class="info-icon-box"></svg>
```
🟡 instead of having those three `if`s, the more stable way is using a classic `if, else if, else` condition - of course the [angular way](https://ultimatecourses.com/blog/angular-ngif-else-then).

Kharonus commented on frontend/src/app/spot/components/breadcrumbs/breadcrumbs.component.html:
```diff
@@ -17,7 +17,15 @@
+          size="small"
+          class="octicon"
+        ></svg>
+        <span *ngIf="crumb.icon !== 'hosting' && crumb.icon !== 'nextcloud-circle' && crumb.icon !== 'cloud'" class="spot-icon spot-icon_{{crumb.icon}}"></span>
```
🔴 I think we do not need the spot crumb anymore. We only have the provider icons here. In addition, do not map the nextcloud icon to a cloud one. we have `op-mark-nextcloud`, which should be used here.

so, in short:
- ondrive and sharepoint go to `cloud`
- nextcloud goes to `op-mark-nextcloud`

Kharonus commented on frontend/src/app/spot/spot.module.ts:
```diff
@@ -19,13 +19,15 @@ import { SpotFormBindingDirective } from './components/form-field/form-binding.d
     ReactiveFormsModule,
     CommonModule,
     A11yModule,
+    CloudIconComponent,
```
🔴 do not import the cloud icon here, but import it in the icons module.

Kharonus commented on frontend/src/global_styles/content/work_packages/shared/_file_picker.sass:
```diff
@@ -56,3 +56,6 @@
       margin-bottom: $spot-spacing-0_25
+
+  .octicon
+    margin-right: $spot-spacing-0_25
```
💡 probably those new classes will carry some of the css overwrites that were previously on spot. If you need help styling, I can get that ;)

Kharonus replied:
Like written above, `spot` carries too much styling to remove the classes from angular components. no style carry over needed.

Kharonus reviewed (changes requested):
I think, we need some polishing here still.

Right now, we loose a lot of icon differentiation, which is not needed.

The usage of the `ngSwitchCase` is a good way to cope with the fact, that we need a separate component for each icon now.

Kharonus committed:
[#667008] introduced more icons for files tab

- reworked ngIf, ngFor and ngSwitch with new syntax in affected files
- amended icon mapping
- enabled attachment icons again
- IMPORTANT: still missing styling fixes

Kharonus commented:
recheck

mereghost reviewed (approved):
I might not have the full angular knowledge to judge this by all angles, or to angle issues here but overall the code looks reasonable.

Kharonus committed:
[#66708] replace icons in attachments section

Kharonus committed:
[#66708] move octicon styling to more generic sass definitions

myabc commented on frontend/src/app/shared/components/storages/openproject-storages.module.ts:
```diff
@@ -92,4 +92,5 @@ import {
     StorageInformationService,
   ],
 })
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
 export class OpenprojectStoragesModule {}
```
The TypeScript decorator needs to directly above the class definition.

You can put the ESLint directive on the same line:

```suggestion
export class OpenprojectStoragesModule {} // eslint-disable-line @typescript-eslint/no-extraneous-class
```

Kharonus replied:
I don't get it yet, what is the benefit from using `eslint-disable-line` over `eslint-disable-next-line`?

myabc replied:
> The TypeScript decorator (`@NgModule`) needs to directly above the class definition.

Kharonus committed:
[#66708] removed unnescesary eslint disabled lines
